app/renderer/components/Inspector/HighlighterRects.js

```diff
@@ -1,20 +1,130 @@
 
+  organizeElements (source) {
+    ELEMENTS_ARRAY = [];
+    OVERLAPS_ARRAY = [];
+    BESTOVERLAPS = [];
+
+    this.copyElementsIntoArray(source, null);
+
+    for (let element of ELEMENTS_ARRAY) {
+      if (this.isMouseOverElement(element.element.attributes)) {
+        OVERLAPS_ARRAY.push(element);
+      }
+    }
+
+    for (let element1 of OVERLAPS_ARRAY) {
+      for (let element2 of ELEMENTS_ARRAY) {
+        if (element2.element !== element1.element
+          && this.isElementOverElement(element1.dimensions, element2.dimensions)
+          && !this.isAncestor(element1.parent, element2.element)) {
+          element1.container = true;
+          break;
+        }
+      }
+    }
+
+    for (let element of OVERLAPS_ARRAY) {
+      const { container, dimensions: { centerX, centerY } } = element;
+      if ((container && distance(centerY, centerX) <= 12) || !container) {
+        BESTOVERLAPS.push(element);
+      }
+    }
+
+    if (BESTOVERLAPS.length > 0) {
+      BESTOVERLAPS.sort(this.compareElements);
+      BESTOVERLAP = BESTOVERLAPS[0];
+    }
+  }
+
+  compareElements (elemA, elemB) {
+    const distA = distance(elemA.dimensions.centerX, elemA.dimensions.centerY);
+    const distB = distance(elemB.dimensions.centerX, elemB.dimensions.centerY);
+    const accessible_A = elemA.element.attributes.accessible;
+    const accessible_B = elemB.element.attributes.accessible;
+
+    if (elemA.container !== elemB.container) { return elemA.container ? -1 : 1; }
+
+    if (distA < distB) { return -1; } else if (distB < distA) { return 1; }
+
+    if (accessible_A !== accessible_B) { return accessible_A ? -1 : 1; }
+
+    return 0;
+  }
+
+  isMouseOverElement ({ x, y, width, height }) {
+    return x <= MOUSE_X
+        && MOUSE_X <= (x + width)
+        && y <= MOUSE_Y
+        && MOUSE_Y <= (y + height);
+  }
+
+  isElementOverElement (element1, element2) {
+    return element1.left <= element2.left
+        && element2.width <= element1.width
+        && element1.top >= element2.top
+        && element2.height <= element1.height;
+  }
+
+  isAncestor (parent, checkElement) {
+    while (parent !== null) {
+      if (parent === checkElement) { return true; }
+
+      for (let elem of ELEMENTS_ARRAY) {
+        if (elem.element === parent) { parent = elem.parent; }
+      }
+    }
+    return false;
+  }
+
+  copyElementsIntoArray (element, prevElement) {
+    if (!element) { return; }
+
+    const {scaleRatio} = this.props;
+    const {x1, y1, x2, y2} = parseCoordinates(element);
+    const left = (x1 / scaleRatio + (this.highlighterXOffset || 0)) || 0;
+    const top = (y1 / scaleRatio) || 0;
+    const width = ((x2 - x1) / scaleRatio) || 0;
+    const height = ((y2 - y1) / scaleRatio) || 0;
+    const centerX = ((x1 + ((x2 - x1) / 2)) / scaleRatio + (this.highlighterXOffset || 0)) || 0;
+    const centerY = ((y1 + ((y2 - y1) / 2)) / scaleRatio) || 0;
+
+    ELEMENTS_ARRAY.push({
```

Comment:
I would rather prefer more explicit variant of this function, where it returns elements array, rather than mutates it implicitly:

```
ELEMENTS_ARRAY.push(...fetchElements());
```